### PR TITLE
Returning the expiry of IdP certs from parseMetadata function 

### PR DESCRIPTION
**Pull request by ukrocks007**
We need the expiry i.e. validTo of the X509 cert received from IdP.
The parseMetadata function has been modified to find and return the validTo of the X509 certificate.

### CODE REVIEW

File: lib/metadata.ts [hunk 1]
```diff
@@ -94,6 +98,22 @@ const parseMetadata = async (idpMeta: string, validateOpts): Promise<Record<stri
 
         if (X509Certificate) {
           ret.thumbprint = thumbprint.calculate(X509Certificate);
+          /**
+           * new crypto.X509Certificate fails with the X509Certificate cert without
+           * -----BEGIN CERTIFICATE-----
+           * and
+           * -----END CERTIFICATE-----
+           */
+          if (X509Certificate.indexOf(BEGIN) != -1 && X509Certificate.indexOf(END) != -1) {
```
Comment (niwsa) [inline lib/metadata.ts:107]: Shouldn't the `else` block be the one that should execute here ?

**Review by deepakprabhakara — APPROVED** [review]

File: lib/metadata.ts (hunk 1 above)
Comment (deepakprabhakara) [reply lib/metadata.ts:107]: @niwsa The if block is looking for missing begin and end blocks

**Review by deepakprabhakara — CHANGES_REQUESTED** [review]
@ukrocks007 Should add tests for this.

File: lib/metadata.ts (hunk 1 above)
Comment (niwsa) [reply lib/metadata.ts:107]: Got it.

File: lib/metadata.ts [hunk 2]
```diff
@@ -94,6 +98,22 @@ const parseMetadata = async (idpMeta: string, validateOpts): Promise<Record<stri
...
+            const { validTo } = new crypto.X509Certificate(X509Certificate);
+            ret.validTo = validTo;
+          } else {
+            /**
+             * Prefixing -----BEGIN CERTIFICATE----- and suffixing -----END CERTIFICATE-----
+             */
+            const { validTo } = new crypto.X509Certificate(`${BEGIN}\n${X509Certificate}${END}`);
```
Comment (niwsa) [inline lib/metadata.ts:114]: @ukrocks007 We can possibly handle the case where one block is missing and possibly avoid duplicate blocks here.

**Comment by ukrocks007** [comment]
> @ukrocks007 Should add tests for this.

Done

File: lib/metadata.ts (hunk 2 above)
Comment (ukrocks007) [reply lib/metadata.ts]: done

**Review by deepakprabhakara — APPROVED** [review]

**Pull request merged**